ui/src/views/Tasks/TaskGroup/index.jsx

```diff
@@ -164,16 +169,98 @@ export default class TaskGroup extends Component {
     searchTerm: null,
   };
 
+  static unsubscribe() {
+    if (!TaskGroup.listener) {
+      return;
+    }
+
+    TaskGroup.listener.unsubscribe();
+    TaskGroup.listener = null;
+  }
+
+  static subscribe({ taskGroupId, subscribeToMore }) {
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId === taskGroupId) {
+      return TaskGroup.listener;
+    }
+
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId !== taskGroupId) {
+      TaskGroup.unsubscribe();
+    }
+
+    const unsubscribe = subscribeToMore({
+      document: taskGroupSubscription,
+      variables: {
+        taskGroupId,
+        subscriptions: [
+          'tasksDefined',
+          'tasksPending',
+          'tasksRunning',
+          'tasksCompleted',
+          'tasksFailed',
+          'tasksException',
+        ],
+      },
+      updateQuery(previousResult, { subscriptionData }) {
+        const { tasksSubscriptions } = subscriptionData.data;
+        // Make sure data is not from another task group which
+        // can happen when a message is in flight and a user searches for
+        // a different task group.
+        const isFromSameTaskGroupId =
+          tasksSubscriptions.taskGroupId === taskGroupId;
+
+        if (
+          !previousResult ||
+          !previousResult.taskGroup ||
+          !isFromSameTaskGroupId
+        ) {
+          return previousResult;
+        }
+
+        const edges = tasks.has(tasksSubscriptions.taskId)
+          ? previousResult.taskGroup.edges.map(edge => {
+              if (tasksSubscriptions.taskId !== edge.node.status.taskId) {
+                return edge;
+              }
+
+              return dotProp.set(edge, 'node', node =>
+                dotProp.set(node, 'status', status =>
+                  dotProp.set(status, 'state', tasksSubscriptions.state)
+                )
+              );
+            })
+          : tasks.set(tasksSubscriptions.taskId) &&
+            previousResult.taskGroup.edges.concat({
+              // eslint-disable-next-line no-underscore-dangle
+              __typename: 'TasksEdge',
+              node: {
+                ...cloneDeep(tasksSubscriptions.task),
```

Comment:
Does this also need to create a `status` property?

Reply:
It already has one.

<img width="569" alt="Screen Shot 2019-04-12 at 2 43 40 PM" src="https://user-images.githubusercontent.com/3766511/56059158-6fae8600-5d31-11e9-94ce-1ff0fa8fa599.png">